GoFSH Best Practices (#48)
FSHSchool/site: content/docs/goFSH/running/_index.md

## GoFSH Best Practices

There are a few best practices that GoFSH users should follow in order to achieve the best results.

* For best results, run GoFSH in `json-only` mode (which is the default) against FHIR definitions that are complete (i.e., have snapshots) and have been validated. This might include:
  * a package downloaded from an IG or FHIR registry (after unzipping it),
  * a package in your local FHIR cache, or

>   * an IG project's `output` folder after it has been built by the IG Publisher.

Reviewer: ```suggestion
  * an IG project's **output** folder after it has been built by the IG Publisher.
```

> * If running GoFSH against a fully built package is not possible, run GoFSH against only the `input` folder, rather than against the entire IG source folder.

Reviewer: ```suggestion
* If running GoFSH against a fully built package is not possible, run GoFSH against only the **input** folder, rather than against the entire IG source folder.
```

>   * In this case be sure to use the proper flags depending on if your source has JSON, XML, or both.

Reviewer: I'd just move this bullet up into the above bullet.

* When possible, avoid running GoFSH on folders that contain:

>   * duplicate files (i.e. definitions that are present in both `input` and `output`),

Reviewer: ```suggestion
  * duplicate files (i.e. definitions that are present in both **input** and **output**),
```

  * incomplete files (i.e. definitions without a snapshot), or

>   * extra files that are not intended to be translated to FHIR Shorthand (e.g, `menu.xml`, files in the input-cache directory).

Reviewer: ```suggestion
  * extra files that are not intended to be translated to FSH (e.g, **menu.xml**, files in the **input-cache** directory).
```
